DataManagementSystem/DB/FileCatalogComponents/SecurityManager/DirectorySecurityManager.py

```diff
@@ -0,0 +1,46 @@
+""" DIRAC FileCatalog Security Manager mix-in class for access check only on the directory level
+"""
+
+__RCSID__ = "$Id$"
+
+import os
+
+from DIRAC import S_OK
+from DIRAC.DataManagementSystem.DB.FileCatalogComponents.SecurityManager.SecurityManagerBase import SecurityManagerBase
+
+class DirectorySecurityManager(SecurityManagerBase):
+
+  def getPathPermissions(self, paths, credDict):
+    """ Get path permissions according to the policy
+    """
+
+    toGet = dict(zip(paths, [[path] for path in paths]))
+    permissions = {}
+    failed = {}
+    while toGet:
+      res = self.db.dtree.getPathPermissions(toGet.keys(), credDict)
+      if not res['OK']:
+        return res
+      for path, mode in res['Value']['Successful'].items():
+        for resolvedPath in toGet[path]:
+          permissions[resolvedPath] = mode
+        toGet.pop(path)
+      for path, error in res['Value']['Failed'].items():
+        if error != 'No such file or directory':
```

Comment:
Shouldn't we use `errno` or `DErrno` here?

Reply:
See #3898